### PR TITLE
Remove bitcoin_quantity

### LINKED ISSUE #1195: Replace bitcoin_quantity usage with rust-bitcoin `Amount` struct
Latest release of rust-bitcoin (0.19.*) provides an `Amount` type. We want to make use of that.

Tasks:
- Upgrade `rust-bitcoin`
- Remove dedicated dependency on `bitcoin_hashes` and use the re-exported `hashes` crate from `rust-bitcoin`
- Remove all usages of `BitcoinQuantity` and replace with `Amount`
- Archive the `bitcoin_quantity` repository

### PR DESCRIPTION
**Pull request by D4nte**
Upgrade rust-bitcoin.
Use some re-exports.
Remove bitcoin_quantity dependency.

Some serializations provided by rust-bitcoin have changed and hence I had to do some custom serializations.

Resolves #1195.

### CODE REVIEW

**Comment by D4nte** [comment]
Unfortunately not finished as we use json strings for assets but rust-bitcoin provide a `u64` serialization for Amount.

File: cnd/src/swap_protocols/actions.rs [hunk 1]
```diff
@@ -11,14 +11,15 @@ pub trait Actions {
...
-    use bitcoin_support::{Address, BitcoinQuantity, Network};
+    use bitcoin_support::{Address, Amount, Network};
     use bitcoin_witness::{PrimedInput, PrimedTransaction};
     use serde::Serialize;
 
     #[derive(Debug, Clone, PartialEq, Serialize)]
     pub struct SendToAddress {
         pub to: Address,
-        pub amount: BitcoinQuantity,
+        #[serde(with = "bitcoin_support::amount::serde::as_sat")]
```
Comment (D4nte) [inline cnd/src/swap_protocols/actions.rs]: Need to add a test here as I assume we use to serialize as a string and now we do an int? Need to double check.

File: cnd/src/http_api/action.rs [hunk 2]
```diff
@@ -34,7 +34,8 @@ pub enum ActionExecutionParameters {
 pub enum ActionResponseBody {
     BitcoinSendAmountToAddress {
         to: bitcoin_support::Address,
-        amount: bitcoin_support::BitcoinQuantity,
+        #[serde(with = "bitcoin_support::amount::serde::as_sat")]
```
Comment (D4nte) [inline cnd/src/http_api/action.rs]: This is strange, we have a serialization for struct `SendToAddress` and for this `ActionResponseBody`.

Comment (D4nte) [inline cnd/src/http_api/action.rs]: Need to add a test here as I assume we use to serialize as a string and now we do an int? Need to double check.

File: cnd/src/http_api/action.rs [hunk 3]
```diff
@@ -339,4 +341,19 @@ mod test {
             r#"{"type":"ethereum-call-contract","payload":{"contract_address":"0x0a81e8be41b21f651a71aab1a85c6813b8bbccf8","gas_limit":"0x1","network":"ropsten"}}"#
         );
     }
+
+    #[test]
+    fn bitcoin_send_amount_to_address_serializes_correctly_to_json_with_none() {
```
Comment (luckysori) [inline cnd/src/http_api/action.rs]: :wrench: The name of this test implies that there is some `None` value but there is not.
```suggestion
    fn bitcoin_send_amount_to_address_serializes_correctly_to_json() {
```

File: cnd/src/http_api/asset.rs [hunk 4]
```diff
@@ -26,6 +26,22 @@ impl HttpAsset {
         serde_json::from_value(parameter).map_err(Error::Serde)
     }
 
+    pub fn parameter_custom_deser<
```
Comment (luckysori) [inline cnd/src/http_api/asset.rs]: :+1: Cool!

File: CHANGELOG.md [hunk 5]
```diff
@@ -14,9 +14,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 - Move config files to standard location based on platform (OSX, Windows, Linux).
 - Align implementation with RFC-002 to use the decision header instead of status codes.
-- Upgrade `btsieve` to `url:2.1`.
+- (btsieve) Upgrade `url` to `2.1`.
```
[the remark this answers is not in the document]
Comment (D4nte) [reply CHANGELOG.md]: It could be relevant for security purposes but then someone could directly check the Cargo.toml history🤷‍♂️

**Pull request merged**